api/app/Models/PoolCandidate.php

```diff
@@ -1178,27 +1177,21 @@ public function computeFinalDecision()
             };
         }
 
-        try {
-            $weight = match ($decision) {
-                FinalDecision::QUALIFIED->name => 10,
-                FinalDecision::QUALIFIED_PENDING->name => 20,
-                FinalDecision::QUALIFIED_PLACED->name => 30,
-                FinalDecision::TO_ASSESS->name => 40,
-                // Set aside some values for assessment steps
-                // Giving a decent buffer to increase max steps
-                FinalDecision::DISQUALIFIED_PENDING->name => 200,
-                FinalDecision::DISQUALIFIED->name => 210,
-                FinalDecision::QUALIFIED_REMOVED->name => 220,
-                FinalDecision::TO_ASSESS_REMOVED->name => 230,
-                FinalDecision::REMOVED->name => 240,
-                FinalDecision::QUALIFIED_EXPIRED->name => 250,
-                default => null
-            };
-        } catch (\UnhandledMatchError $e) {
```

Comment:
Can we still log an error if `default` is matched?  I don't have much experience with PHP pattern matching.  I think we still want to be notified if this falls through, though.

Reply:
Yeah, added the log back in [add log for default match](https://github.com/GCTC-NTGC/gc-digital-talent/pull/11938/commits/d809c647353f02d518c63df783ce08ca53b6f4ca)